pkg/cmd/roachtest/tests/restore.go

```diff
@@ -953,34 +945,94 @@ type restoreSpecs struct {
 	backup   backupSpecs
 	timeout  time.Duration
 	tags     []string
+
+	t        test.Test
+	c        cluster.Cluster
+	testName string
 }
 
-func (sp *restoreSpecs) computeName(full bool) string {
-	return "restore" + sp.backup.String(full) + sp.hardware.String(full)
+func (sp *restoreSpecs) computeName(prefix string, verbose bool) string {
+	if prefix != "" {
+		prefix = "/" + prefix
+	}
+	return "restore" + prefix + sp.backup.String(verbose) + sp.hardware.String(verbose)
 }
 
-func (sp *restoreSpecs) restoreCmd() string {
-	return fmt.Sprintf(`./cockroach sql --insecure -e "RESTORE FROM %s IN %s AS OF SYSTEM TIME '%s'"`,
-		sp.backup.fullBackupDir, sp.backup.backupCollection(), sp.backup.aost)
+func (sp *restoreSpecs) restoreCmd(target, opts string) string {
+	return fmt.Sprintf(`./cockroach sql --insecure -e "RESTORE %s FROM %s IN %s AS OF SYSTEM TIME '%s' %s"`,
+		target, sp.backup.fullBackupDir, sp.backup.backupCollection(), sp.backup.aost, opts)
 }
 
-func (sp *restoreSpecs) getRuntimeSpecs(ctx context.Context, t test.Test, c cluster.Cluster) {
+func (sp *restoreSpecs) getRuntimeSpecs(
+	ctx context.Context, t test.Test, c cluster.Cluster, testName string,
+) {
+	sp.t = t
+	sp.c = c
+	sp.testName = testName
+
 	var aost string
-	conn := c.Conn(ctx, t.L(), 1)
+	conn := sp.c.Conn(ctx, sp.t.L(), 1)
 	err := conn.QueryRowContext(ctx, sp.backup.getAostCmd()).Scan(&aost)
-	require.NoError(t, err)
+	require.NoError(sp.t, err)
 	sp.backup.aost = aost
 }
 
-func (sp *restoreSpecs) run(ctx context.Context, c cluster.Cluster) error {
-	if err := c.RunE(ctx, c.Node(1), sp.restoreCmd()); err != nil {
-		return errors.Wrapf(err, "full test specs: %s", sp.computeName(true))
+// run executes the restore, where target injects a restore target into the restore command.
+// Examples:
+// - "DATABASE tpce" will execute a database restore on the tpce cluster.
+// - "" will execute a cluster restore.
+func (sp *restoreSpecs) run(ctx context.Context, target string) error {
+	return sp.c.RunE(ctx, sp.c.Node(1), sp.restoreCmd(target, ""))
+}
+
+func (sp *restoreSpecs) runDetached(ctx context.Context, target string) (jobspb.JobID, error) {
```

Comment:
I think you can now delete the old runRestoreDetached?

Reply:
yup, in a follow up PR, i plan delete all the old infra and bank tests :D